### PR TITLE
Add support for topic propagation IRC -> Slack

### PR DESCRIPTION
**Pull request by marcuswanner**

### CODE REVIEW

**Comment by marcuswanner** [comment]
I'll fix the lint issues above and update the pull req later today.

**Comment by ekmartin** [comment]
Thanks! I'm a bit unsure if this is good default behavior though. Would you necessarily want the Slack topic to match the IRC topic? For most of the channels that I use slack-irc for, the topic is set to something like "Bridge to #channel on Freenode" or similar, to explain what's happening.

File: lib/bot.js [hunk 1]
```diff
@@ -102,6 +102,11 @@ class Bot {
       this.sendToSlack(author, to, formattedText);
     });
 
+    this.ircClient.on('topic', (channel, topic, nick) => {
+      this.sendToSlack(this.nickname, channel, `*${nick}* changed the topic`);
```
Comment (ekmartin) [inline lib/bot.js]: Sending the changed topic to Slack as a message is definitely good default behavior though. Maybe the message should include the channel name too? `<username> changed the topic of the IRC channel #chat to "<topic>"`

**Comment by marcuswanner** [comment]
I'll add a config option if you specify name + bahavior for it.

**Comment by jpouellet** [comment]
Slack seems to have a 250 char max on titles, whereas many IRC servers do not. It would be nice to get a warning or something on IRC if the change on slack fails for that reason.

**Comment by marcuswanner** [comment]
What if I just truncated it?

**Comment by jpouellet** [comment]
I would prefer if truncation would produce some warning instead of silently truncating.

**Comment by ekmartin** [comment]
Sorry for the extremely late answer. I'm going to close this, as I think it would be tricky to define a sane default behavior for it. Hiding it behind a configuration option is definitely possible, but with an already quite large configuration file I'd prefer to keep it as small as possible. There's also the question about what directions should be synced: Would it make sense to be able to choose if you wanted this to happen the other way around too (i.e. from Slack to IRC as well)?

Sending a message to Slack whenever the topic changes on IRC is a good idea though, and something I don't think a configuration option would be needed for. I don't think the direction is much of a problem here either, and what you've already done here is a good solution, which I'd be happy to merge if you wanted to send a separate pull request for that.

**Pull request closed without merging**